src/status_im/common/bottom_sheet/view.cljs

```diff
@@ -96,9 +102,45 @@
         bottom                            (if selected-item-smaller-than-sheet?
                                             (+ sheet-height bottom-margin)
                                             (:bottom insets))
-        sheet-max-height                  (- window-height (:top insets))
+        sheet-max-height                  (- window-height
+                                             (:top insets)
+                                             kb-height
+                                             (when platform/ios?
+                                               keyboard-vertical-offset))
         content-padding-bottom            (or padding-bottom-override
                                               (+ (:bottom insets) bottom-margin))]
+    (rn/use-mount
+     (fn []
+       (when (zero? kb-height)
+         (async-storage/get-item :kb-default-height
+                                 (fn [height]
+                                   (set-kb-height (utils.number/parse-int height 0)))))
+       (when (nil? @keyboard-show-listener)
+         (reset! keyboard-show-listener
+           (.addListener rn/keyboard
+                         "keyboardDidShow"
+                         (fn [event]
+                           (let [height (oops/oget event "endCoordinates.height")]
+                             (set-kb-height height)
+                             (when (zero? kb-height)
+                               (async-storage/set-item! :kb-default-height (str height))))))))
+       (when (nil? @keyboard-layout-listener)
+         (reset! keyboard-layout-listener
+           (.addListener
+            rn/keyboard
+            "keyboardWillChangeFrame"
+            (fn [event]
+              (let [height (oops/oget event "endCoordinates.height")]
+                (set-kb-height height)
+                (async-storage/set-item! :kb-default-height (str height)))))))))
```

Comment:
Can't we use `react-native.hooks/use-keyboard` ?

Reply:
I think we can, however there are a few glitches that happen when iOS changes keyboard layouts from ABC to Emoji. For example, the bottom-sheet sorta flickers as the keyboard changes height, this is what `keyboardWillChangeFrame` is used for in this case. 

Some other glitches also happen on Android, but we don't have access to `keyboardWithChangeFrame`, so I'm looking into another way of handling these layout changes.

Reply:
> so I'm looking into another way of handling these layout changes.

@seanstrom are the Android keyboard dimensions changing?

Could you please share what are those found issues?

Reply:
@ulisesmac yeah the keyboard dimensions can change on iOS and Android when someone switches between the emoji picker and alphabet keyboards. Android also seems to have more keyboards that will also take up more space.

Here's a recording of one of the issues that happen with changing keyboard layouts. The video should show how the bottom-sheet is pushed up towards the top of the screen, but the max-height of the bottom-sheet is recalculated to not go past the top safe-area, though that calculation happens too late and there's a small flicker.

https://github.com/status-im/status-mobile/assets/2845768/41138d6f-d1c7-49a1-9724-8a49a5baa1cc

I think ideally the bottom-sheet would push to the top and not flicker, and I have something that might work by using `flex-shrink` on the entire bottom-sheet. However it's not quite perfect either because there's some flickers when the keyboard decreases in size.

https://github.com/status-im/status-mobile/assets/2845768/2c93edd2-de83-4ddf-a5ca-03eb2008c3f1

Reply:
That's true, but IMO it looks good enough since the modal is properly showed.

I'm not sure if this may help:
https://docs.swmansion.com/react-native-reanimated/docs/device/useAnimatedKeyboard/

we should test if it reports the changed size, if so, we should first implement it as a separate component, since many of our UIs will benefit from it (ofc, I know it's totally out of scope of this issue).

Reply:
Yup yup I think I'll create a separate issue that tries to fix these smaller glitches. And thanks for the reference to that animated keyboard 🙏 , I'll take a look and hopefully that will save me some effort with trying to configure flexbox rules. Happy to make a separate component when I get something solid figured out 🙌